distributed/comm/utils.py

```diff
@@ -58,15 +57,10 @@ def _to_frames():
             logger.exception(e)
             raise
 
-    if OFFLOAD_THRESHOLD and allow_offload:
-        try:
-            msg_size = sizeof(msg)
-        except RecursionError:
-            msg_size = math.inf
-    else:
-        msg_size = 0
-
-    if allow_offload and OFFLOAD_THRESHOLD and msg_size > OFFLOAD_THRESHOLD:
+    # sizeof(msg) can raise RecursionError; _to_frames() can cause msgpack to
+    # raise ValueError (improperly; should be RecursionError).
+    # The two are not necessarily going to happen at the same depth.
+    if allow_offload and OFFLOAD_THRESHOLD and sizeof(msg) > OFFLOAD_THRESHOLD:
```

Comment:
Why is this new behavior better than before?

if sizeof hits the recursion error, we're now screwed. But IIUC we would've been screwed anywhow since in this case msgpack would've run into a recursion error?

Reply:
> we would've been screwed anywhow since in this case msgpack would've run into a recursion error?

Correct.
It's better than before because it removes a special edge case and simplifies the code. It also emits a more correct exception in both cases.

### main:
- 0~429 recursion depth: works
- 430~511: works (special case handling). `safe_sizeof()` elsewhere logs a warning and defaults to 1MB.
- 512+: ValueError

### this PR:
- 0~429: works
- 430+: RecursionError

### msgpack 1.0.6/Windows, both on main and on this PR:
- 0~255: works
- 256+: ValueError

Reply:
Wait. I measured it wrong.
I'm now getting that sizeof() fails at depth 140. Which makes the special case quite important.
I'll rework the PR.